### PR TITLE
Adds a searchbar to the Traitor uplink

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

 Ports ss220-space/Paradise/pull/848. It adds a search bar to the traitor/nukie uplink, allowing them to search for objects in their uplink to buy. It searches the name, description, price, and if its a hijack item and shows the results. The way this is done is we get all the items in one list, and then we can filter them out. Its slightly sketchy but it works.

## Why It's Good For The Game
Huge QOL for traitors/nukies, I always forget where some gear is that I want to buy.

## Images of changes

image
image
image
image

## Testing

Bought a lot of shit, messed around with the tabs, worked as intended. Searching for name, description, cost, and if its a hijack item all worked. Switching between the tabs clears the search bar, as intended.

## Changelog
:cl:
add: The traitor uplink now has a search bar
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/Uplink.js [hunk 1]
```diff
@@ -64,11 +71,38 @@ const ItemsPage = (_properties, context) => {
...
+        is_hijack);
+    });
+    return flow([
+      filter(item => item?.name), // Make sure it has a name
+      searchText && filter(EquipmentSearch), // Search for anything
+      sortBy(item => item?.name), // Sort by name
+    ])(cat);
+  };
+  const handleSearch = (value) => {
+    if(value === ""){
```
Comment (Sirryan2002) [inline tgui/packages/tgui/interfaces/Uplink.js:99]:
```suggestion
    if(value === "") {
```

File: tgui/packages/tgui/interfaces/Uplink.js [hunk 2]
```diff
@@ -32,15 +33,21 @@ export const Uplink = (props, context) => {
           <Tabs.Tab
             key="PurchasePage"
             selected={tabIndex === 0}
-            onClick={() => setTabIndex(0)}
+            onClick={() => {
+                setTabIndex(0);
+                setSearchText("");
```
Comment (Sirryan2002) [inline tgui/packages/tgui/interfaces/Uplink.js:38]: why bother setting search text to a blank string when its only used for the equipment page?

Comment (Contrabang) [reply tgui/packages/tgui/interfaces/Uplink.js:38]: The search is also used for the exploitable information page

**Review by S34NW — APPROVED** [review]
Should be okay, although sirryan is better with TGUI than me. Deconflict and should be okay.

**Review by SteelSlayer — APPROVED** [review]

**Comment by SteelSlayer** [comment]
Once this is deconflicted, it can be merged.

**Pull request merged**